### PR TITLE
Fix BlockParameter parsing for filters

### PR DESCRIPTION
**Pull request by LukaszRozmej**
Closes #3265 PR

## Changes:
Remove obsolete (and buggy) deserialization.

## Types of changes

What types of changes does your code introduce?

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [ ] Yes
- [x] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by MicahZoltu — APPROVED** [review]
Thanks!

File: src/Nethermind/Nethermind.JsonRpc/Data/BlockParameterConverter.cs [hunk 1]
```diff
@@ -128,16 +128,22 @@ public override BlockParameter ReadJson(JsonReader reader, Type objectType, Bloc
...
+        public static BlockParameter GetBlockParameter(string? value)
+        {
             switch (value)
             {
-                case "":
+                case null:
+                case { } empty when string.IsNullOrWhiteSpace(empty):
                 case { } latest when latest.Equals("latest", StringComparison.InvariantCultureIgnoreCase):
                     return BlockParameter.Latest;
                 case { } latest when latest.Equals("earliest", StringComparison.InvariantCultureIgnoreCase):
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.JsonRpc/Data/BlockParameterConverter.cs:142]: [you_can_ignore_it] latest and renaming?

**Review by MarekM25 — CHANGES_REQUESTED** [review]
Could you add any test or explain why we don't need it?

**Comment by LukaszRozmej** [comment]
We removed unused code, moved to better, actually tested one.

**Review by MarekM25 — APPROVED** [review]

**Pull request merged**